servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/RequestTagExtractor.java

```diff
@@ -32,15 +32,30 @@ private static String getRequestMethod(HttpRequestMetaData req) {
     }
 
     private static String getHttpUrl(HttpRequestMetaData req) {
-        return (req.scheme() == null ? "http" : req.scheme()) + "://" +
-            (req.effectiveHostAndPort() == null ? "localhost:8080" : req.effectiveHostAndPort())
-            + req.rawPath()
+        return req.path()
             + (req.rawQuery() == null ? "" : "?" + req.rawQuery());
     }
 
     static Span reportTagsAndStart(SpanBuilder span, HttpRequestMetaData httpRequestMetaData) {
         span.setAttribute("http.url", getHttpUrl(httpRequestMetaData));
         span.setAttribute("http.method", getRequestMethod(httpRequestMetaData));
+        span.setAttribute("http.target", getHttpUrl(httpRequestMetaData));
+        span.setAttribute("http.route", httpRequestMetaData.rawPath());
+        span.setAttribute("http.flavor", httpRequestMetaData.version().major() + "."
+            + httpRequestMetaData.version().minor());
```

Comment:
sorry, didn't know if you finished or not and pushed https://github.com/apple/servicetalk/pull/2536/commits/27a25e2bde99c1f639c72a63c8c36093f19234ed
since this format is specific to OTEL, let's keep it private for now. If in the future we will have another case, we can add a new method on `HttpProtocolVersion`.

Reply:
no problem, let's keep it with the private method to cache the version for now.